lib/services/user_service.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:async';
+import 'dart:math';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:uuid/uuid.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import '../models/user.dart';
+
+User currentUser;
+UserService userService;
+
+class UserService {
```

Comment:
i would prefer abstracting this class. e.g. `FirebaseUserService` implements `UserService` and we have kind of factory method for easily switch from firebase to other datastore.